testsuite/forge/src/backend/k8s_deployer/deployer.rs

```diff
@@ -0,0 +1,457 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    FORGE_DEPLOYER_IMAGE_TAG, FORGE_DEPLOYER_SERVICE_ACCOUNT_NAME,
+    FORGE_DEPLOYER_VALUES_ENV_VAR_NAME, FORGE_INDEXER_DEPLOYER_DOCKER_IMAGE_REPO,
+    FORGE_TESTNET_DEPLOYER_DOCKER_IMAGE_REPO,
+};
+use crate::{maybe_create_k8s_resource, K8sApi, ReadWrite, Result};
+use k8s_openapi::api::{
+    batch::v1::Job,
+    core::v1::{ConfigMap, Namespace, ServiceAccount},
+    rbac::v1::RoleBinding,
+};
+use kube::{
+    api::{ObjectMeta, PostParams},
+    ResourceExt,
+};
+use std::{collections::BTreeMap, fmt, sync::Arc};
+
+#[derive(Clone, Copy)]
+pub enum ForgeDeployerType {
+    Indexer,
+    Testnet,
+}
```

Comment:
We should also get rid of type, this doesnt seem to serve a purpose other than 

1) get_name 
2) selecting the image

Get name doesnt need it, and the full image path / tag should be passed in from the caller not decided in the deployer